File: python/Linked List Cycle.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def hasCycle(self, head):
        try:
            slow = head
            fast = head.next
            while slow is not fast:
                slow = slow.next
                fast = fast.next.next
            return True
        except AttributeError:
            return False


def create_linked_list(values, pos):
    if not values:
        return None

    nodes = [ListNode(val) for val in values]
    for i in range(len(nodes) - 1):
        nodes[i].next = nodes[i + 1]

    if pos < len(nodes):
        nodes[-1].next = nodes[pos]  # Create a cycle

    return nodes[0]


# Example usage
values = [1, 2]
pos = 0

head = create_linked_list(values, pos)

sol = Solution()
result = sol.hasCycle(head)
print(result)

File: JAVA/java/LinkedListCycle.java
class ListNode {
    int val;
    ListNode next;

    ListNode(int x) {
        val = x;
        next = null;
    }
}

public class LinkedListCycle {
    public boolean hasCycle(ListNode head) {
        try {
            ListNode slow = head;
            ListNode fast = head.next;
            while (slow != fast) {
                slow = slow.next;
                fast = fast.next.next;
            }
            return true;
        } catch (NullPointerException e) {
            return false;
        }
    }

    public static ListNode createLinkedList(int[] values, int pos) {
        if (values == null || values.length == 0) {
            return null;
        }

        ListNode[] nodes = new ListNode[values.length];
        for (int i = 0; i < values.length; i++) {
            nodes[i] = new ListNode(values[i]);
        }

        for (int i = 0; i < values.length - 1; i++) {
            nodes[i].next = nodes[i + 1];
        }

        if (pos < values.length) {
            nodes[values.length - 1].next = nodes[pos];  // Create a cycle
        }

        return nodes[0];
    }

    public static void main(String[] args) {
        int[] values = {1, 2};
        int pos = 0;

        ListNode head = createLinkedList(values, pos);

        LinkedListCycle solution = new LinkedListCycle();
        boolean result = solution.hasCycle(head);
        System.out.println(result);
    }
}

